torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -715,6 +713,12 @@ PjRtComputationClient::ExecuteReplicated(
                   ->Execute(std::move(argument_handles), execute_options,
                             returned_futures)
                   .value();
+
+    (*returned_futures)[0].OnReady(
+        std::move([timed, op = std::move(op)](xla::Status unused) mutable {
+          timed.reset();
```

Comment:
If `timed` is a shared_ptr already, can you just move & capture, insteaed of explicitly resetting?

Reply:
I was confused by this at first too. The trick is, we have to ensure the `TimedSection` ends before the `Operation` does, otherwise we may allow `WaitDeviceOps` to proceed before we register the `ExecuteReplicated` metrics. The destruction order of captured variables is undefined.

Adding a comment since we both questioned this.